okapi-core/src/test/java/org/folio/okapi/InstallTest.java

```diff
@@ -1514,4 +1519,353 @@ public void installTenantInitVersion2JobError(TestContext context) {
     tModule.stop().onComplete(context.asyncAssertSuccess());
   }
 
+  @Test
+  public void parallelMigration(TestContext context) {
+    JsonObject moda = createAsyncInitModule("mod-a-1.0.0");
+    moda.getJsonArray("provides")
+        .add(new JsonObject()
+            .put("id", "a")
+            .put("version", "1.0")
+        );
+    given()
+        .header("Content-Type", "application/json")
+        .body(moda.encode()).post("/_/proxy/modules").then().statusCode(201);
+
+    JsonObject modb = createAsyncInitModule("mod-b-1.0.0");
+    modb.getJsonArray("provides")
+        .add(new JsonObject()
+            .put("id", "b")
+            .put("version", "1.0")
+        );
+    given()
+        .header("Content-Type", "application/json")
+        .body(modb.encode()).post("/_/proxy/modules").then().statusCode(201);
+
+    JsonObject mid = createAsyncInitModule("mod-m-1.0.0");
+    mid.getJsonArray("requires")
+        .add(new JsonObject()
+            .put("id", "a")
+            .put("version", "1.0")
+        )
+        .add(new JsonObject()
+            .put("id", "b")
+            .put("version", "1.0")
+        );
+    given()
+        .header("Content-Type", "application/json")
+        .body(mid.encode()).post("/_/proxy/modules").then().statusCode(201);
+
+    final String tenant = "roskilde";
+    createTenant(context, tenant);
+    deployAsyncInitModule(context, moda.getString("id"), portModule);
+    deployAsyncInitModule(context, modb.getString("id"), portModule);
+    deployAsyncInitModule(context, mid.getString("id"), portModule);
+    ModuleTenantInitAsync tModule = new ModuleTenantInitAsync(vertx, portModule);
+    tModule.start().onComplete(context.asyncAssertSuccess());
```

Comment:
```
    Async start = context.async();
    tModule.start().onComplete(context.asyncAssertSuccess(x -> start.complete()));
    start.await();
```
Otherwise the following post might run before start has completed.

Reply:
sure